From 2998028946bb793e6463cb9cb302173ba4f3e75c Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 12:58:40 -0400
Subject: [ethernet] qlge: Fix vlan netdev features

Message-id: <1408625927-3701-10-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88536
O-Subject: [PATCH RHEL 7.1 09/16] qlge: Fix vlan netdev features.
Bugzilla: 1089357
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>

Bug 1089357 (https://bugzilla.redhat.com/show_bug.cgi?id=1089357)

net commit id 51bb352f15595f2dee42b599680809de3d08999d

vlan gets the same netdev features except vlan filter.

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlge/qlge_main.c b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
index e07cbcd..aca96ed 100644
--- a/drivers/net/ethernet/qlogic/qlge/qlge_main.c
+++ b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
@@ -4767,6 +4767,8 @@ static int qlge_probe(struct pci_dev *pdev,
        NETIF_F_RXCSUM;
  ndev->features = ndev->hw_features;
  ndev->vlan_features = ndev->hw_features;
+ /* vlan gets same features (except vlan filter) */
+ ndev->vlan_features &= ~NETIF_F_HW_VLAN_CTAG_FILTER;
 
  if (test_bit(QL_DMA64, &qdev->flags))
   ndev->features |= NETIF_F_HIGHDMA;
-- 
1.7.1